src/lib/frontend/d_cnf.ml

```diff
@@ -97,42 +101,40 @@ module Cache = struct
   let store_var ind v =
     store_sy ind (Sy.var v)
 
-  let find_ty ind =
-    HT.find ae_ty_ht ind
+  let find_ty id =
+    HT.find ae_ty_ht (Id id)
 
-  let fresh_ty ?(is_var = true) ?(name = None) () =
+  let fresh_ty ?(is_var = true) ?(id = None) () =
```

Comment:
I think you want `?id` here and `~id` when `fresh_ty` is called.

(Currently `fresh_ty` has a `?id:Uid.t option` optional argument, simply using `?id` would make it a `?id:Uid.t` argument)